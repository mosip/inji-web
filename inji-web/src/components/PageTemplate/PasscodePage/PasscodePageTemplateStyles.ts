export const PasscodePageTemplateStyles = {
    overlay: "fixed inset-0 backdrop-blur-sm bg-black bg-opacity-40 flex flex-col items-center justify-center z-50 h-screen",
    container: "rounded-2xl bg-white flex flex-col items-center justify-start relative w-[90%] sm:w-[85%] md:w-[90%] h-fit max-h-[99vh] sm:max-h-[98vh] overflow-y-hidden shadow-iw-pin-page-container",
    contentWrapper: "flex flex-col items-center justify-start w-full mt-44 sm:mt-48 relative z-10 pb-0 sm:pb-8",
    titleContainer: "text-center space-y-5 relative w-full max-w-lg px-4 sm:px-0",
    title: "text-2xl-plus font-semibold text-iw-darkGrayishBlue",
    description: "text-iw-textTertiary text-lg font-normal mx-8",
    subTitle: {
        container: 'flex items-start gap-3 justify-center mx-4 px-2',
        backArrowButton: 'cursor-pointer mt-2 sm:mt-1',
        content: 'text-iw-textTertiary text-base font-medium',
    },
    errorAndContentContainer: "flex flex-col bg-white sm:rounded-lg sm:shadow-iw-pin-page-container items-center z-20 w-full max-w-lg mt-8 mb-4 mx-auto border-t border-iw-grayishBlue mx-8",
    warningTextBorder: "relative sm:hidden pin-page-warning-text-border min-w-full items-center justify-center mt-1 sm:mt-3 md:mt-5",
    warningTextContainer: "w-full items-center justify-center flex",
    warningText: "text-iw-textTertiary text-center text-sm sm:text-base px-4 sm:px-8 pt-4 sm:pt-6 max-w-xl",
    errorContainer: "bg-iw-pink50 w-full px-3 sm:px-5 py-3 border-t sm:rounded-t-lg",
    errorContentWrapper: "flex items-start justify-between gap-2 max-w-xl mx-auto",
    errorText: "text-sm text-iw-darkRed break-words whitespace-normal",
    errorTextContainer: "flex-1 pr-2",
    closeButtonContainer: "flex-shrink-0",
    closeButton: "cursor-pointer",
    closeIcon: "min-w-[26px] min-h-[26px] mt-1.5 sm:mt-2 text-iw-darkRed",
    bottomBorder: "pin-page-warning-text-border w-full mt-2 sm:mt-3 md:mt-5",
    contentContainer: "w-full px-4 sm:px-8 py-3 sm:py-5 md:py-7 space-y-4 flex flex-col items-center",
}