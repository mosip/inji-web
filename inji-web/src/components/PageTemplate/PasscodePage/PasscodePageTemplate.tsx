import React from 'react';
import {BackgroundDecorator} from '../../Common/BackgroundDecorator';
import {CrossIconButton} from '../../Common/Buttons/CrossIconButton';
import {BackArrowButton} from "../../Common/Buttons/BackArrowButton";
import {PasscodePageTemplateStyles} from "./PasscodePageTemplateStyles";

interface PasscodePageTemplateProps {
    title: string;
    subtitle: string;
    error?: string | null;
    onErrorClose?: () => void;
    content: React.ReactNode;
    testId: string;
    onBack?: () => void;
}

export const PasscodePageTemplate: React.FC<PasscodePageTemplateProps> = ({
                                                                              title,
                                                                              subtitle,
                                                                              error,
                                                                              onErrorClose,
                                                                              content,
                                                                              testId,
                                                                              onBack
                                                                          }) => {
    return (
        <div
            data-testid={`${testId}-page`}
            className={PasscodePageTemplateStyles.overlay}
        >
            <div className={PasscodePageTemplateStyles.container}>
                <BackgroundDecorator testId={`backdrop-${testId}`}/>

                <div className={PasscodePageTemplateStyles.contentWrapper}>
                    <div className={PasscodePageTemplateStyles.titleContainer}>
                        <h1
                            className={PasscodePageTemplateStyles.title}
                            data-testid={`title-${testId}`}
                        >
                            {title}
                        </h1>
                        {
                            onBack ? (
                                <div className={PasscodePageTemplateStyles.subTitle.container}>
                                    <BackArrowButton
                                        onClick={onBack}
                                        btnClassName={PasscodePageTemplateStyles.subTitle.backArrowButton}
                                    />
                                    <p
                                        className={PasscodePageTemplateStyles.subTitle.content}
                                        data-testid="subtitle-reset-passcode"
                                    >
                                        {subtitle}
                                    </p>
                                </div>
                            ) : <p
                                className={PasscodePageTemplateStyles.description}
                                data-testid="passcode-description"
                            >
                                {subtitle}
                            </p>
                        }
                    </div>

                    <div
                        className={PasscodePageTemplateStyles.errorAndContentContainer}
                        data-testid="passcode-inputs-container"
                    >
                        {error && (
                            <div
                                className={PasscodePageTemplateStyles.errorContainer}
                                data-testid="error-passcode"
                            >
                                <div className={PasscodePageTemplateStyles.errorContentWrapper}>
                                    <div className={PasscodePageTemplateStyles.errorTextContainer}>
                                        <span className={PasscodePageTemplateStyles.errorText}>
                                          {error}
                                        </span>
                                    </div>
                                    {onErrorClose && (
                                        <div className={PasscodePageTemplateStyles.closeButtonContainer}>
                                            <CrossIconButton
                                                onClick={onErrorClose}
                                                btnClassName={PasscodePageTemplateStyles.closeButton}
                                                iconClassName={PasscodePageTemplateStyles.closeIcon}
                                            />
                                        </div>
                                    )}
                                </div>
                            </div>
                        )}

                        <div className={PasscodePageTemplateStyles.contentContainer}>
                            {content}
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};