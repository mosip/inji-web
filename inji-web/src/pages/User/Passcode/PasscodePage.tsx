import React, {Fragment, useEffect, useState} from 'react';
import {useNavigate} from 'react-router-dom';
import {api, MethodType} from '../../../utils/api';
import {useCookies} from 'react-cookie';
import {SolidButton} from '../../../components/Common/Buttons/SolidButton';
import {useTranslation} from 'react-i18next';
import {useUser} from '../../../hooks/User/useUser';
import {PasscodeInput} from '../../../components/Common/Input/PasscodeInput';
import {navigateToUserHome} from "../../../utils/navigationUtils";
import {PasscodePageStyles} from './PasscodePageStyles';
import {ROUTES} from "../../../utils/constants";
import {PasscodePageTemplate} from "../../../components/PageTemplate/PasscodePage/PasscodePageTemplate";
import {TertiaryButton} from "../../../components/Common/Buttons/TertiaryButton";

export const PasscodePage: React.FC = () => {
    const {t} = useTranslation('PasscodePage');
    const navigate = useNavigate();
    const [error, setError] = useState<string | null>(null);
    const [loading, setLoading] = useState<boolean>(false);
    const [wallets, setWallets] = useState<any[] | undefined>(undefined);
    const [cookies] = useCookies(['XSRF-TOKEN']);
    const [passcode, setPasscode] = useState<string[]>(Array(6).fill(''));
    const [confirmPasscode, setConfirmPasscode] = useState<string[]>(
        Array(6).fill('')
    );
    const {saveWalletId} = useUser();

    const fetchWallets = async () => {
        try {
            const response = await fetch(api.fetchWallets.url(), {
                method: MethodType[api.fetchWallets.methodType],
                headers: {
                    ...api.fetchWallets.headers(),
                    'X-XSRF-TOKEN': cookies['XSRF-TOKEN']
                },
                credentials: 'include'
            });

            const responseData = await response.json();

            if (!response.ok) {
                throw responseData;
            }

            setWallets(responseData);
        } catch (error) {
            console.error('Error occurred while fetching Wallets:', error);
            setError(t('error.fetchWalletsError'));
        }
    };

    useEffect(() => {
        const fetchWalletDetails = async () => {
            await fetchWallets();
        };
        fetchWalletDetails();
    }, []);

    useEffect(() => {
        const handleStorageChange = (e: StorageEvent) => {
            if (e.key === 'walletId') {
                fetchWallets();
            }
        };

        window.addEventListener('storage', handleStorageChange);
        return () => window.removeEventListener('storage', handleStorageChange);
    }, []);

    const unlockWallet = async (walletId: string, pin: string) => {
        if (!walletId) {
            setError(t('error.walletNotFoundError'));
            navigate(ROUTES.PASSCODE);
            throw new Error('Wallet not found');
        }

        try {
            const response = await fetch(api.fetchWalletDetails.url(walletId), {
                method: MethodType[api.fetchWalletDetails.methodType],
                headers: {
                    ...api.fetchWalletDetails.headers(),
                    'X-XSRF-TOKEN': cookies['XSRF-TOKEN']
                },
                credentials: api.fetchWalletDetails.credentials,
                body: JSON.stringify({walletPin: pin})
            });

            const responseData = await response.json();
            if (!response.ok) {
                setError(t('error.incorrectPasscodeError'));
                throw responseData;
            }
            saveWalletId(walletId)
        } catch (error) {
            throw error;
        }
    };

    const createWallet = async () => {
        try {
            const pin = passcode.join('');
            const confirmPin = confirmPasscode.join('');

            if (pin !== confirmPin) {
                setError(t('error.passcodeMismatchError'));
                throw new Error('Pin and Confirm Pin mismatch');
            }

            const response = await fetch(api.createWalletWithPin.url(), {
                method: 'POST',
                headers: {
                    ...api.createWalletWithPin.headers(),
                    'X-XSRF-TOKEN': cookies['XSRF-TOKEN']
                },
                credentials: 'include',
                body: JSON.stringify({
                    walletPin: pin,
                    confirmWalletPin: confirmPasscode.join(''),
                    walletName: null
                })
            });

            if (!response.ok) {
                const errorData = await response.json();
                setError(
                    `${t('error.createWalletError')}: ${
                        errorData.errorMessage ?? t('unknown-error')
                    }`
                );
                throw errorData;
            }

            const createdWallet = await response.json();
            await unlockWallet(createdWallet.walletId, pin);

            setWallets([{walletId: createdWallet.walletId}]);
        } catch (error) {
            throw error;
        }
    };

    const handleSubmit = async () => {
        setError('');
        setLoading(true);

        try {
            if (isUserCreatingWallet()) {
                await createWallet();
            } else {
                const walletId = wallets ? wallets[0].walletId : undefined
                const formattedPasscode = passcode.join('');

                if (formattedPasscode.length !== 6) {
                    setError(t('error.passcodeLengthError'));
                    setLoading(false);
                    return;
                }

                await unlockWallet(walletId, formattedPasscode);
            }
            await navigateToUserHome(navigate);
        } catch (error) {
            console.error(
                'Error occurred while setting up Wallet or loading user profile',
                error
            );
        } finally {
            setLoading(false);
        }
    };

    const isButtonDisabled =
        passcode.includes('') ||
        (isUserCreatingWallet() && confirmPasscode.includes(''));

    function isUserCreatingWallet() {
        return wallets?.length === 0;
    }

    const pageTitle = isUserCreatingWallet() ? t('setPasscode') : t('enterPasscode');
    const pageSubtitle = isUserCreatingWallet() ? t('setPasscodeDescription') : t('enterPasscodeDescription');

    function renderForgotPasscodeButton() {
        const handleForgotPasscode = () =>
            navigate(
                ROUTES.USER_RESET_PASSCODE,
                {
                    state: {
                        walletId: wallets ? wallets[0].walletId : undefined
                    }
                }
            );

        return <div className={PasscodePageStyles.forgotPasscodeContainer}>
            <TertiaryButton onClick={handleForgotPasscode} title={t('forgotPasscode') + "?"}
                            testId={"forgot-passcode"} className={PasscodePageStyles.forgotPasscodeButton}/>
        </div>;
    }

    function renderPasscodeInput(label: string, value: string[], onChange: (values: string[]) => void, testId: string) {
        return <PasscodeInput label={label} value={value} onChange={onChange} testId={testId}/>;
    }

    const renderContent = () => {
        return (
            <div className={PasscodePageStyles.contentContainer}>
                {wallets &&
                    <Fragment>
                        {<div className={PasscodePageStyles.inputWrapper}>
                            {renderPasscodeInput(t('enterPasscodeLabel'), passcode, setPasscode, "passcode")}

                            {isUserCreatingWallet() &&
                                renderPasscodeInput(t('confirmPasscodeLabel'), confirmPasscode, setConfirmPasscode, "confirm-passcode")
                            }
                        </div>
                        }
                        {
                            !isUserCreatingWallet() && renderForgotPasscodeButton()
                        }
                        <div className={PasscodePageStyles.buttonContainer}>
                            <SolidButton
                                fullWidth={true}
                                testId="btn-submit-passcode"
                                onClick={handleSubmit}
                                title={loading ? t('submitting') : t('submit')}
                                disabled={isButtonDisabled}
                                className={isButtonDisabled ? PasscodePageStyles.disabledButton : ''}
                            />
                        </div>
                    </Fragment>
                }

            </div>
        );
    };

    return (
        <PasscodePageTemplate
            title={pageTitle}
            subtitle={pageSubtitle}
            error={error}
            onErrorClose={() => setError(null)}
            content={renderContent()}
            contentTestId={"passcode-inputs-container"}
            testId="passcode"
        />
    );
};
