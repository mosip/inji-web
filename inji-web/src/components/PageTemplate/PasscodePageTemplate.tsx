import React from 'react';
import {PasscodePageStyles} from '../../pages/User/Passcode/PasscodePageStyles';
import {BackgroundDecorator} from '../Common/BackgroundDecorator';
import {CrossIconButton} from '../Common/Buttons/CrossIconButton';
import {ResetPasscodePageStyles} from "../../pages/User/ResetPasscode/ResetPasscodePageStyles";
import {BackArrowButton} from "../Common/Buttons/BackArrowButton";

interface PasscodePageTemplateProps {
    title: string;
    subtitle: string;
    error?: string | null;
    onErrorClose?: () => void;
    content: React.ReactNode;
    testId: string;
    onBack?: () => void;
}

export const PasscodePageTemplate: React.FC<PasscodePageTemplateProps> = ({
                                                                              title,
                                                                              subtitle,
                                                                              error,
                                                                              onErrorClose,
                                                                              content,
                                                                              testId,
                                                                              onBack
                                                                          }) => {
    return (
        <div
            data-testid={testId}
            className={PasscodePageStyles.pageOverlay}
        >
            <div className={PasscodePageStyles.container}>
                <BackgroundDecorator
                    logoSrc={require('../../assets/Logomark.png')}
                    logoAlt="Inji Web Logo"
                    logoTestId="logo-inji-web"
                />

                <div className={PasscodePageStyles.contentWrapper}>
                    <div className={PasscodePageStyles.titleContainer}>
                        <h1
                            className={PasscodePageStyles.title}
                            data-testid="title-passcode"
                        >
                            {title}
                        </h1>
                        {
                            onBack ? (
                                <div className={ResetPasscodePageStyles.subHeader}>
                                    <BackArrowButton
                                        onClick={onBack}
                                        btnClassName={ResetPasscodePageStyles.backArrowButton}
                                    />
                                    <p
                                        className={ResetPasscodePageStyles.subtitle}
                                        data-testid="subtitle-reset-passcode"
                                    >
                                        {subtitle}
                                    </p>
                                </div>
                            ) : <p
                                className={PasscodePageStyles.description}
                                data-testid="passcode-description"
                            >
                                {subtitle}
                            </p>
                        }
                    </div>

                    <div
                        className={PasscodePageStyles.passcodeContainer}
                        data-testid="passcode-inputs-container"
                    >
                        {error && (
                            <div
                                className={PasscodePageStyles.errorContainer}
                                data-testid="error-passcode"
                            >
                                <div className={PasscodePageStyles.errorContentWrapper}>
                                    <div className={PasscodePageStyles.errorTextContainer}>
                                        <span className={PasscodePageStyles.errorText}>
                                          {error}
                                        </span>
                                    </div>
                                    {onErrorClose && (
                                        <div className={PasscodePageStyles.closeButtonContainer}>
                                            <CrossIconButton
                                                onClick={onErrorClose}
                                                btnClassName={PasscodePageStyles.closeButton}
                                                iconClassName={PasscodePageStyles.closeIcon}
                                            />
                                        </div>
                                    )}
                                </div>
                            </div>
                        )}

                        <div className={PasscodePageStyles.inputContainer}>
                            {content}
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};