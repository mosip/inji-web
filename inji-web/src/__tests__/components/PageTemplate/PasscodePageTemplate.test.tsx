import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import {PasscodePageTemplate} from '../../../components/PageTemplate/PasscodePage/PasscodePageTemplate';

describe('PasscodePageTemplate Component', () => {
    const mockOnErrorClose = jest.fn();
    const mockOnBack = jest.fn();
    const mockTestId = 'test';
    const mockTitle = 'Test Title';
    const mockSubtitle = 'Test Subtitle';
    const mockError = 'Test Error';
    const mockContent = <div data-testid="content">Test Content</div>;

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('renders with all elements correctly', () => {
        render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                error={mockError}
                onErrorClose={mockOnErrorClose}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
            />
        );

        expect(screen.getByTestId('test-page')).toBeInTheDocument();
        expect(screen.getByTestId('backdrop-test')).toBeInTheDocument();
        expect(screen.getByTestId('title-test')).toHaveTextContent(mockTitle);
        expect(screen.getByTestId('test-description')).toHaveTextContent(mockSubtitle);
        expect(screen.getByTestId('error-test')).toBeInTheDocument();
        expect(screen.getByTestId('btn-close-icon-container')).toBeInTheDocument();
        expect(screen.getByTestId('content')).toBeInTheDocument();
    });

    test('renders without error when error is null', () => {
        render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                error={null}
                onErrorClose={mockOnErrorClose}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
            />
        );

        expect(screen.queryByTestId('error-passcode')).not.toBeInTheDocument();
    });

    test('renders with back button when onBack is provided', () => {
        render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
                onBack={mockOnBack}
            />
        );

        // Back button should be rendered
        expect(screen.getByTestId('btn-back-arrow-container')).toBeInTheDocument();
        expect(screen.getByTestId('icon-back-arrow')).toBeInTheDocument();
        expect(screen.getByTestId('subtitle-test')).toHaveTextContent(mockSubtitle);

        // Regular subtitle should not be present
        expect(screen.queryByTestId('passcode-description')).not.toBeInTheDocument();
    });

    test('calls onErrorClose when close button is clicked', () => {
        render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                error={mockError}
                onErrorClose={mockOnErrorClose}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
            />
        );

        fireEvent.click(screen.getByTestId('btn-close-icon-container'));

        expect(mockOnErrorClose).toHaveBeenCalledTimes(1);
    });

    test('calls onBack when back button is clicked', () => {
        render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
                onBack={mockOnBack}
            />
        );

        fireEvent.click(screen.getByTestId('btn-back-arrow-container'));
        expect(mockOnBack).toHaveBeenCalledTimes(1);
    });

    test('renders without cross button when onErrorClose is not provided', () => {
        render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                error={mockError}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
            />
        );

        expect(screen.getByTestId('error-test')).toBeInTheDocument();
        expect(screen.queryByTestId('cross-icon-button')).not.toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const {asFragment} = render(
            <PasscodePageTemplate
                title={mockTitle}
                subtitle={mockSubtitle}
                error={mockError}
                onErrorClose={mockOnErrorClose}
                content={mockContent}
                contentTestId={"content-test"}
                testId={mockTestId}
                onBack={mockOnBack}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});