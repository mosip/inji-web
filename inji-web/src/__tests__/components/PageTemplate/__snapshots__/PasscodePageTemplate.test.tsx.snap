// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasscodePageTemplate Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 backdrop-blur-sm bg-black bg-opacity-40 flex flex-col items-center justify-center z-50 h-screen"
    data-testid="test-page"
  >
    <div
      class="rounded-2xl bg-white flex flex-col items-center justify-start relative w-[90%] sm:w-[85%] md:w-[90%] h-fit overflow-y-auto shadow-iw-pin-page-container"
    >
      <div
        class="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
        data-testid="backdrop-test"
      >
        <div
          class="absolute top-[116px]"
          style="top: 116px;"
        >
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="backdrop-test-circle-0"
            style="width: 96px; height: 96px; border-width: 1px; border-color: rgba(228, 231, 236, 0.8); top: calc(50% - 48px); left: calc(50% - 48px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="backdrop-test-circle-1"
            style="width: 192px; height: 192px; border-width: 1px; border-color: rgba(228, 231, 236, 0.7000000000000001); top: calc(50% - 96px); left: calc(50% - 96px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="backdrop-test-circle-2"
            style="width: 288px; height: 288px; border-width: 1px; border-color: rgba(228, 231, 236, 0.6000000000000001); top: calc(50% - 144px); left: calc(50% - 144px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="backdrop-test-circle-3"
            style="width: 384px; height: 384px; border-width: 1px; border-color: rgba(228, 231, 236, 0.5); top: calc(50% - 192px); left: calc(50% - 192px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="backdrop-test-circle-4"
            style="width: 480px; height: 480px; border-width: 1px; border-color: rgba(228, 231, 236, 0.4); top: calc(50% - 240px); left: calc(50% - 240px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="backdrop-test-circle-5"
            style="width: 576px; height: 576px; border-width: 1px; border-color: rgba(228, 231, 236, 0.30000000000000004); top: calc(50% - 288px); left: calc(50% - 288px);"
          />
          <div
            class="flex items-center justify-center"
            data-testid="logo-inji-web-container"
          >
            <img
              alt="Inji Web Logo"
              data-testid="logo-inji-web"
              src="Logomark.png"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-start w-full mt-[180px] sm:mt-[200px] relative z-10 pb-0 sm:pb-8"
      >
        <div
          class="text-center space-y-5 relative w-full max-w-[500px] px-4 sm:px-0"
        >
          <h1
            class="text-2xl-plus font-semibold text-iw-darkGrayishBlue"
            data-testid="title-test"
          >
            Test Title
          </h1>
          <div
            class="flex items-center gap-3 justify-center mx-4 px-2"
          >
            <button
              class="cursor-pointer"
              data-testid="btn-back-arrow-container"
              type="button"
            >
              <img
                alt="Back Arrow"
                class=""
                data-testid="icon-back-arrow"
                src="BackArrowIcon.svg"
              />
            </button>
            <p
              class="text-iw-textTertiary text-base font-medium"
              data-testid="subtitle-reset-passcode"
            >
              Test Subtitle
            </p>
          </div>
        </div>
        <div
          class="flex flex-col bg-white sm:rounded-lg sm:shadow-iw-pin-page-container items-center z-20 w-full max-w-[500px] mt-8 mb-4 mx-auto border-t border-iw-grayishBlue mx-8 sm:my:0"
          data-testid="passcode-inputs-container"
        >
          <div
            class="bg-iw-pink50 w-full px-3 sm:px-5 py-3 border-t sm:rounded-t-lg"
            data-testid="error-passcode"
          >
            <div
              class="flex items-start justify-between gap-2 max-w-[560px] mx-auto"
            >
              <div
                class="flex-1 pr-2"
              >
                <span
                  class="text-sm text-iw-darkRed break-words whitespace-normal"
                >
                  Test Error
                </span>
              </div>
              <div
                class="flex-shrink-0"
              >
                <button
                  class="cursor-pointer"
                  data-testid="btn-close-icon-container"
                  type="button"
                >
                  <svg
                    class="min-w-[26px] min-h-[26px] mt-1.5 sm:mt-2 text-iw-darkRed"
                    data-testid="icon-close"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M6.2 6.2a1 1 0 011.4 0L12 10.6l4.4-4.4a1 1 0 111.4 1.4L13.4 12l4.4 4.4a1 1 0 01-1.4 1.4L12 13.4l-4.4 4.4a1 1 0 11-1.4-1.4L10.6 12 6.2 7.6a1 1 0 010-1.4z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="w-full px-4 sm:px-8 py-3 sm:py-5 md:py-7 space-y-4 flex flex-col items-center"
          >
            <div
              data-testid="content"
            >
              Test Content
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
